### PR TITLE
Extract boilerplate code into Routed base classes

### PR DESCRIPTION
**Pull request by RobbeSneyders**
As mentioned in #1588, we're repeating a lot of boilerplate across the middlewares. This PR extracts this boilerplate into shared base classes. Feel free to propose better naming for them :slightly_smiling_face: 

I had to revert the order of Middleware - Api - Operation to Operation - Api - Middleware due to dependencies in the typing.

### CODE REVIEW

File: connexion/middleware/routing.py [hunk 1]
```diff
@@ -6,61 +6,36 @@
 from starlette.types import ASGIApp, Receive, Scope, Send
```
Comment (RobbeSneyders) [inline connexion/middleware/routing.py:6]: Only changed the order in this file.

File: connexion/middleware/swagger_ui.py [hunk 2]
```diff
@@ -21,56 +21,6 @@
 _original_scope: ContextVar[Scope] = ContextVar("SCOPE")
```
Comment (RobbeSneyders) [inline connexion/middleware/swagger_ui.py:21]: Only changed the order in this file.

File: connexion/middleware/routing.py (hunk 1 above)
Comment (RobbeSneyders) [inline connexion/middleware/routing.py:6]: Only changed the order in this file and moved the `ROUTING_CONTEXT`.

File: connexion/middleware/abstract.py [hunk 3]
```diff
@@ -12,3 +26,120 @@ def add_api(
...
+    its operations.
+
+    The RoutingMiddleware adds the operation_id to the ASGI scope. This middleware registers its
+    operations by operation_id at startup. At request time, the operation is fetched by an
+    operation_id lookup.
+    """
+
+    @property
+    @abc.abstractmethod
+    def api_cls(self) -> t.Type[API]:
```
Comment (Ruwann) [inline connexion/middleware/abstract.py:98]: Possible to do similar as in RouteAPI to avoid property?

File: connexion/middleware/security.py [hunk 4]
```diff
@@ -304,8 +198,47 @@ def _get_verification_fn(self):
...
+            next_app=self.next_app,
+            security_handler_factory=self.security_handler_factory,
+        )
+
+
+class SecurityMiddleware(RoutedMiddleware[SecurityAPI]):
+    """Middleware to check if operation is accessible on scope."""
+
+    @property
+    def api_cls(self) -> t.Type[SecurityAPI]:
```
Comment (Ruwann) [inline connexion/middleware/security.py:240]: see base class

**Review by Ruwann — APPROVED** [review]

**Pull request merged**